tornado/web.py

```diff
@@ -2809,6 +2809,10 @@ def validate_absolute_path(self, root: str, absolute_path: str) -> Optional[str]
                 self.redirect(self.request.path + "/", permanent=True)
                 return None
             absolute_path = os.path.join(absolute_path, self.default_filename)
+            if os.path.join(self.root, self.path, self.default_filename) != absolute_path:
```

Comment:
I don't think this works on windows. `abspath` converts all forward slashes to backslashes; `os.path.join` does not. I'm not sure if we have enough test coverage to catch this.

Reply:
I added a replace statement now... I think it works on Windows, although someone should try it. I don't have a windows computer